core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -881,6 +884,16 @@ public void withStartupAttempts(int attempts) {
         this.startupAttempts = attempts;
     }
 
+    @Override
+    public void close() {
+        stop();
+    }
+
+    public SELF withCustomizer(Consumer<CreateContainerCmd> hook) {
```

Comment:
Your post clarified the API a lot for me and now I agree, pre/post seems to introduce confusing semantics here.

From a general API naming standpoint, `withConfigurationCustomizer` and `withConfigModifier` both _sound_ good, but isn't it actually more a `withCreateContainerCmdModifier`? Sounds awkward and very verbose, but kind of in line with the docker-java API (which seems appropriate, since docker-java does leak through this method after all).